modules/phase_field/src/ics/SmoothCircleFromFileIC.C

```diff
@@ -0,0 +1,92 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SmoothCircleFromFileIC.h"
+//#include "MooseRandom.h"
+
+template <>
+InputParameters
+validParams<SmoothCircleFromFileIC>()
+{
+  InputParameters params = validParams<SmoothCircleBaseIC>();
+  params.addClassDescription(
+      "Multiple smooth circles read from a text file");
+  params.addRequiredParam<FileName>("file_name", "File containing circle centers and radii");
+  params.addParam<unsigned int>("header_length", 0, "Number of lines to skip at top of file");
+
+  return params;
+}
+
+SmoothCircleFromFileIC::SmoothCircleFromFileIC(const InputParameters & parameters)
+  : SmoothCircleBaseIC(parameters)
+{
+  FileName file_name = getParam<FileName>("file_name");
+  unsigned int header_length = getParam<unsigned int>("header_length");
+  std::vector<Real> data;
+  std::ifstream inFile;
+  unsigned int n_circles = 0;
+
+  MooseUtils::checkFileReadable(file_name);
+  inFile.open(file_name.c_str());
+
+  for (unsigned int i = 0; i < header_length; ++i)
+    inFile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  std::string line;
+  while (std::getline(inFile, line))
+    ++n_circles;
+
+  inFile.close();
+
+  data.resize(n_circles * 4);
+
+  inFile.open(file_name.c_str());
+
+  for (unsigned int i = 0; i < header_length; ++i)
+    inFile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  for (unsigned int i = 0; i < data.size(); ++i)
+    inFile >> data[i];
+
+  inFile.close();
+
+  for (unsigned int i = 0; i < data.size(); ++i)
```

Comment:
Also this loop is wrong. You can see it now that you've applied the patch. I don't think you need the loop at all.